xbmc/utils/URIUtils.h

```diff
@@ -332,6 +332,9 @@ class URIUtils
    */
   static bool UpdateUrlEncoding(std::string &strFilename);
 
+  static CURL AddCredentials(const CURL& url);
+  static CURL AddCredentials(CURL&& url);
```

Comment:
out of curiousity. why not only declare the `const CURL&` function and define it like this?
```cpp
CURL URIUtils::AddCredentials(const CURL& url)
{
  CURL authUrl{url};
  if (CPasswordManager::GetInstance().IsURLSupported(authUrl) && authUrl.GetUserName().empty())
    CPasswordManager::GetInstance().AuthenticateURL(authUrl);

  return authUrl;
}
```
should be the same when we never call `URIUtils::AddCredentials(std::move(something));` anywhere or directly with a temporary

Reply:
one copy avoided, got it, thank you